openshift/dockerfiles/cli.Dockerfile

```diff
@@ -0,0 +1,33 @@
+ARG GO_BUILDER=brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.22
+ARG RUNTIME=registry.access.redhat.com/ubi9/ubi-minimal:latest@sha256:c0e70387664f30cd9cf2795b547e4a9a51002c44a4a86aa9335ab030134bf392
+
+FROM $GO_BUILDER AS builder
+
+ARG TKN_PAC_VERSION=nightly
+WORKDIR /go/src/github.com/openshift-pipelines/pipelines-as-code
+COPY . .
+RUN set -e; for f in patches/*.patch; do echo ${f}; [[ -f ${f} ]] || continue; git apply ${f}; done
+ENV GODEBUG="http2server=0"
+RUN go build -mod=vendor -tags disable_gcp -v  \
+    -ldflags "-X github.com/openshift-pipelines/pipelines-as-code/pkg/params/version.Version=${TKN_PAC_VERSION}" \
+    -o /tmp/tkn-pac ./cmd/tkn-pac
```

Comment:
why don't you use the make target instead of using directly your go files?

Reply:
I think this is mostly because they are coming from downstream, and look very similar to any other `openshift-pipelines` component's `Dockerfile` 🙃 (aka at some point, I was thinking of generating it)

Reply:
well if we want to have this upstream you may use the same target as what we have upstream, having multiple way to compile the binaries across the repository is not really good code hygiene

Reply:
I see make sense

actually this https://github.com/openshift-pipelines/pipelines-as-code/blob/main/Dockerfile and PR dockerfile almost same but base images are different 
@vdemeester can we try with existing dockerfile and lets make components to point out to this existing Dockerfile :thinking: